jumanji/environments/routing/macvrp/env.py

```diff
@@ -0,0 +1,600 @@
+
+    - reward: jax array (float32)
+        this global reward is provided to each agent. The reward is equal to the negative sum of the
+        distances between consecutive nodes at the end of the episode over all agents. All time
+        penalties are also added to the reward.
+
+    - observation and state:
+        the observation and state variable types are defined in:
+        jumanji/environments/routing/macvrp/types.py
+
+    [1] Zhang et al. (2020). "Multi-Vehicle Routing Problems with Soft Time Windows: A
+    Multi-Agent Reinforcement Learning Approach".
+    """
+
+    """Instantiates a `MACVRP` environment.
+
+        Args:
+            num_nodes: number of city nodes in the environment. Defaults to 20.
+            max_capacity: maximum capacity of the vehicle. Defaults to 30.
+            max_demand: maximum demand of each node. Defaults to 10.
+            reward_fn: `RewardFn` whose `__call__` method computes the reward of an environment
+                transition. The function must compute the reward based on the current state,
+                the chosen action, the next state and whether the action is valid.
+                Implemented options are [`SparseReward`]. Defaults to `DenseReward`.
+            viewer: `Viewer` used for rendering. Defaults to `CVRPViewer` with "human" render mode.
+        """
+
+    def __init__(
+        self,
+        generator: Optional[Generator] = None,
+        reward_fn: Optional[RewardFn] = None,
+        viewer: Optional[Viewer] = None,
+    ):
+        """
+        Instantiates a `MACVRP` environment.
+
+        Args:
+            generator: `Generator` whose `__call__` instantiates an environment instance.
+                Implemented options are [`UniformRandomGenerator`].
+                Defaults to `UniformRandomGenerator` with `num_customers=20` and `num_vehicles=2`.
+            reward_fn: `RewardFn` whose `__call__` method computes the reward of an environment
+                transition. The function must compute the reward based on the current state
+                and whether the environment is done.
+                Implemented options are [`DenseReward`]. Defaults to `DenseReward`.
+            viewer: `Viewer` used for rendering. Defaults to `MACVRPViewer` with "human" render
+                mode.
+        """
+
+        # Create generator used for generating new environments
+        self._generator = generator or UniformRandomGenerator(
+            num_customers=6,
+            num_vehicles=2,
+        )
+
+        self._max_capacity = self._generator._max_capacity
+        self._map_max = self._generator._map_max
+        self._customer_demand_max = self._generator._customer_demand_max
+        self._max_start_window = self._generator._max_start_window
+        self._max_end_window = self._generator._max_end_window
+        self._time_window_length = self._generator._time_window_length
+        self._early_coef_rand = self._generator._early_coef_rand
```

Comment:
I find `_early_coef_rand` to not be very meaningful, is it the min and max value for the early_coeff ?

Reply:
Removed. It is not really used anymore.